Tests/CI fail due to a regression in requests 2.32.0

See e.g. https://github.com/bokeh/bokeh/actions/runs/9209200886/job/25333399799?pr=13908#step:8:416:
```py
self = <urllib3.poolmanager.PoolManager object at 0x7fb3caebcce0>
request_context = {'block': False, 'cert_reqs': 'CERT_REQUIRED', 'host': 'test.socket', 'maxsize': 10, ...}

    def connection_from_context(self, request_context):
        """
        Get a :class:`urllib3.connectionpool.ConnectionPool` based on the request context.
    
        ``request_context`` must at least contain the ``scheme`` key and its
        value must be a key in ``key_fn_by_scheme`` instance variable.
        """
        scheme = request_context["scheme"].lower()
        pool_key_constructor = self.key_fn_by_scheme.get(scheme)
        if not pool_key_constructor:
>           raise URLSchemeUnknown(scheme)
E           urllib3.exceptions.URLSchemeUnknown: Not supported URL scheme http+unix
```

--- a/conda/environment-test-3.10.yml
+++ b/conda/environment-test-3.10.yml
@@ -41,7 +41,7 @@ dependencies:
   - pytest-html
   - pytest-xdist
   - pytest-timeout
-  - requests >=1.2.3
+  - requests >=1.2.3,!=2.32.* # see https://github.com/bokeh/bokeh/issues/13910
   - urllib3 <2 # see https://github.com/bokeh/bokeh/issues/13152
   - scipy
   - pooch # required in scipy.datasets

--- a/conda/environment-test-3.11.yml
+++ b/conda/environment-test-3.11.yml
@@ -41,7 +41,7 @@ dependencies:
   - pytest-html
   - pytest-xdist
   - pytest-timeout
-  - requests >=1.2.3
+  - requests >=1.2.3,!=2.32.* # see https://github.com/bokeh/bokeh/issues/13910
   - urllib3 <2 # see https://github.com/bokeh/bokeh/issues/13152
   - scipy
   - pooch # required in scipy.datasets

--- a/conda/environment-test-3.12.yml
+++ b/conda/environment-test-3.12.yml
@@ -41,7 +41,7 @@ dependencies:
   - pytest-html
   - pytest-xdist
   - pytest-timeout
-  - requests >=1.2.3
+  - requests >=1.2.3,!=2.32.* # see https://github.com/bokeh/bokeh/issues/13910
   - urllib3 <2 # see https://github.com/bokeh/bokeh/issues/13152
   - scipy
   - pooch # required in scipy.datasets

--- a/conda/environment-test-minimal-deps.yml
+++ b/conda/environment-test-minimal-deps.yml
@@ -40,7 +40,7 @@ dependencies:
   - pytest-html
   - pytest-xdist
   - pytest-timeout
-  - requests >=1.2.3
+  - requests >=1.2.3,!=2.32.* # see https://github.com/bokeh/bokeh/issues/13910
   - urllib3 <2 # see https://github.com/bokeh/bokeh/issues/13152
   - selenium 4.2
 

